src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
> distance_exp() literally just takes a node IDs, XORs it with our node ID and then returns the xor distance between them (160 - number_of_leading_zeros)

Distances can differ in bits beyond the shared prefix bits. That's relevant if you want to compare multiple items relatively close to each other.